File: gcj/gcj2016/qual/a/a.exs
defmodule Main do
  def main do
    tt = IO.gets("")
    {tt, _} = Integer.parse(tt)
    for t <- 1..tt do
      n = IO.gets("")
      {n, _} = Integer.parse(n)
      IO.puts "Case ##{t}: #{solve(n)}"
    end
  end

  def solve(n) do
    if n == 0 do
      :INSOMNIA
    else
      seen = MapSet.new([])
      ans = func(n, 1, seen)
    end
  end

  def func(n, cnt, seen) do
    if MapSet.size(seen) == 10 do
      (cnt-1)*n
    else
      tmp = Integer.digits(n*cnt)
      tmp = MapSet.new(tmp)
      seen = MapSet.union(seen,tmp)
      func(n, cnt+1, seen)
    end    
  end
end

Main.main

File: gcj/gcj2016/qual/a/a.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

def solve(n):
    seen = set()
    cnt = 1
    ans = None
    while True:
        val = n * cnt
        seen.update( str(val) )
        # print("val, seen:", val, seen)

        if len(seen) == 10:
            ans = val
            break
        if cnt == 1000000:
            ans = "INSOMNIA"
            break

        cnt += 1

    return ans

# for n in range(int(1e6)):
#     ans = solve(n)
#     print("n, ans: ", n, ans)

T=int(input())
for testcase in range(T):
    n = int(input())
    ans = solve(n)
    print("Case #{}: {}".format(testcase+1, ans))
